### DISCUSSION TITLE
[Question][Hw4] Set semantic

### DISCUSSION THREAD

**Discussion by john-mai-2605** [prosyslab-classroom/cs348-information-security#568] [Q&A]
Is there a way to pull out an element from a set randomly, without changing the set to another Data Structure?
I read the Set documentation and seems like it is all deterministic.

**Comment by ilchu** [comment]
From what I could gather, it really is all deterministic, so we would have to cast it into array or list. More info here: [https://stackoverflow.com/questions/5782932/how-can-i-randomly-select-an-element-in-ocaml](https://stackoverflow.com/questions/5782932/how-can-i-randomly-select-an-element-in-ocaml)

**Comment by john-mai-2605** [comment]
Thank you. I will close the issue :).
